Modules/Devices/src/Devices.Infrastructure/Persistence/Repository/TiersRepository.cs

```diff
@@ -83,6 +83,6 @@ public async Task<Tier> GetBasicTierAsync(CancellationToken cancellationToken)
 
     public async Task<IEnumerable<Tier>> FindByIds(IEnumerable<TierId> tierIds, CancellationToken cancellationToken)
     {
-        return await _tiersDbSet.Where(t => tierIds.Contains(t.Id)).ToListAsync(cancellationToken) ?? throw new NotFoundException(nameof(Tier));
+        return await _tiers.Where(t => tierIds.Contains(t.Id)).ToListAsync(cancellationToken) ?? throw new NotFoundException(nameof(Tier));
```

Comment:
Just noticed: `ToListAsync` never returns null. So you can remove the exception.

```suggestion
        return await _tiers.Where(t => tierIds.Contains(t.Id)).ToListAsync(cancellationToken) ?? throw new NotFoundException(nameof(Tier));
```